aicsdeformation/processing.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from functools import partial
+import itertools
+import logging
+from multiprocessing.dummy import Pool
+import numpy as np
+import openpiv.process
+import openpiv.scaling
+import openpiv.validation
+import openpiv.filters
+from typing import Dict, List, NamedTuple, Set, Tuple, Union
+from tqdm import tqdm
+
+from .types import Displacement
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+# openpiv reference:
+# https://openpiv.readthedocs.io/en/latest/src/tutorial.html
+
+
+class DisplacementFromParameters(NamedTuple):
+    displacement: Displacement
+    parameters: Dict[str, Union[int, float]]
+
+
+class ErrorFromParameters(NamedTuple):
+    error: ValueError
+    parameters: Dict[str, Union[int, float]]
+
+
+def calculate_displacement(
+    frames: Tuple[np.ndarray] = None,
+    frame_a: np.ndarray = None,
+    frame_b: np.ndarray = None,
+    window_size: int = 18,
+    overlap: int = 4,
+    dt: float = 0.003,
+    search_area_size: int = 20,
+    sig2noise_method: str = "peak2peak"
+) -> Displacement:
+    """
+    Generate base displacement. This is primarly used for determining signal : noise between two frames before
+    additional processing is conducted.
+
+    :param frames: A tuple of two numpy.ndarray frames to find displacements for.
+        This will be expanded to frame_a and frame_b if not None.
+    :param frame_a: A single frame to be used to find displacements for.
+    :param frame_b: A single frame to be used to find displacements for.
+
+    Look to OpenPIV for details on how parameters interact with the actual displacement generation.
+    https://openpiv.readthedocs.io/en/latest/src/tutorial.html
+    """
+    # Expand frames if required
+    if frames:
+        frame_a, frame_b = (*frames,)
```

Comment:
Is the comma in (*frames,) just to protect against frames only containing a single frame?